File: Solutions/1758/1758.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1758.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-11-29 10:33:03
***************************************************************************************************/


int minOperations(char * s){
    int ret1=0, ret2=0;
    char ta='1', tb='0';
    for (int i=0;i<strlen(s);i++) {
        if (s[i] != ta) ret1++;
        if (s[i] != tb) ret2++;
        ta = ta == '1' ? '0' : '1';
        tb = tb == '1' ? '0' : '1';
    }
    return ret1 > ret2 ? ret2 : ret1;
}

File: Solutions/1758/1758.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/11/29 17:03
# File: 1758.py
# Desc: 


class Solution:
    def minOperations(self, s: str) -> int:
        ret1 = ret2 = 0
        ta = '0'
        tb = '1'
        for i in s:
            if i != ta:
                ret1 += 1
            ta = '0' if ta == '1' else '1'
            if i != tb:
                ret2 += 1
            tb = '0' if tb == '1' else '1'
        return min(ret2, ret1)
